x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/lib/is_rate.ts

```diff
@@ -8,16 +8,16 @@
 import { has } from 'lodash';
 import {
   MetricsUIAggregation,
-  MetricsUIAggregationRT,
+  hasAggregations,
   ESSumBucketAggRT,
-  ESTermsWithAggregationRT,
+  hasSnapshotTermsWithAggregation,
   ESDerivativeAggRT,
   ESBasicMetricAggRT,
 } from '@kbn/metrics-data-access-plugin/common';
 import { SnapshotCustomMetricInput } from '../../../../../common/http_api';
 
 export const isMetricRate = (metric: MetricsUIAggregation | undefined): boolean => {
-  if (!MetricsUIAggregationRT.is(metric)) {
+  if (hasAggregations(metric) || metric === undefined) {
     return false;
   }
```

Comment:
Q: Isn't `hasAggregations(metric)` redundant, considering the `metric` is already defined as `MetricUIAggregation`? Perhaps we could leave just the `undefined` check here?

Reply:
I left only the undefined check 👍